semantic-kernel/concepts/vector-store-connectors/out-of-the-box-connectors/postgres-connector.md

```diff
@@ -62,91 +86,420 @@ using Npgsql;
+        }
+    }
+
+    private static string? TryGetUsernameFromToken(string jwtToken)
+    {
+        // Split the token into its parts (Header, Payload, Signature)
+        var tokenParts = jwtToken.Split('.');
+        if (tokenParts.Length != 3)
+        {
+            return null;
+        }
+
+        // The payload is the second part, Base64Url encoded
+        var payload = tokenParts[1];
+
+        // Add padding if necessary
+        payload = AddBase64Padding(payload);
+
+        // Decode the payload from Base64Url
+        var decodedBytes = Convert.FromBase64String(payload);
+        var decodedPayload = Encoding.UTF8.GetString(decodedBytes);
+
+        // Parse the decoded payload as JSON
+        var payloadJson = JsonSerializer.Deserialize<JsonElement>(decodedPayload);
+
+        // Try to get the username from 'upn', 'preferred_username', or 'unique_name' claims
+        if (payloadJson.TryGetProperty("upn", out var upn))
+        {
+            return upn.GetString();
+        }
+        else if (payloadJson.TryGetProperty("preferred_username", out var preferredUsername))
+        {
+            return preferredUsername.GetString();
+        }
+        else if (payloadJson.TryGetProperty("unique_name", out var uniqueName))
+        {
+            return uniqueName.GetString();
+        }
+
+        return null;
+    }
+
+    private static string AddBase64Padding(string base64) => (base64.Length % 4) switch
+    {
+        2 => base64 + "==",
+        3 => base64 + "=",
+        _ => base64,
+    };
+}
+```
+
+Now you can use the `UseEntraAuthentication` method to set up the connection string for the Postgres connector:
+
+```csharp
+var dataSourceBuilder = new NpgsqlDataSourceBuilder(connectionString);
+dataSourceBuilder.UseEntraAuthentication();
+dataSourceBuilder.UseVector();
+using var datasource = dataSourceBuilder.Build();
+
+var vectorStore = new PostgresVectorStore(dataSource);
+```
+
+By default, the `UseEntraAuthentication` method uses the [DefaultAzureCredential](/dotnet/api/azure.identity.defaultazurecredential) to authenticate with Azure AD. You can also provide a custom `TokenCredential` implementation if needed.
+
 ::: zone-end
 ::: zone pivot="programming-language-python"
 
-## Coming soon
+## Getting started
+
+Install semantic kernel with the Postgres extras, which includes the [Postgres client](https://github.com/Postgres/Postgres-client).
+
+```cli
+pip install semantic-kernel[postgres]
+```
+
+You can then create a vector store instance using the `PostgresStore` class.
+You can pass a `psycopg_pool` [AsyncConnectionPool](https://www.psycopg.org/psycopg3/docs/api/pool.html#psycopg_pool.AsyncConnectionPool) directly,
+or use the `PostgresSettings` to create a connection pool from environment variables.
+
+```python
+
+from semantic_kernel.connectors.memory.Postgres import PostgresStore, PostgresSettings
 
-More info coming soon.
+settings = PostgresSettings()
+with settings.create_connection_pool() as pool:
```

Comment:
as far as I can see in the code, this is not a context manager, and it is def a async method
```suggestion
pool = await settings.create_connection_pool()
vector_store = PostgresStore(pool=pool)
```

Reply:
btw, I would be open to making this a context manager, I think that makes sense, we would have to do a PR for that though!

Reply:
This is fixed; the pool is a context manager, I was just using it incorrectly with out the async awaits.